media/media.go

```diff
@@ -143,21 +163,82 @@ func (sm svgManager) getFullPath(mediaPath string) string {
 	return filepath.Join(sm.baseDir, mediaPath)
 }
 
-func (sm svgManager) ensureDownloaded(mediaPath string) error {
+func (sm svgManager) ensureDownloaded(mediaPath string, c color.Color) (string, error) {
+	// use the colour as a directory to separate different colours of SVG's
+	customizedMediaPath := path.Join(colorToHex6(c), mediaPath)
+
 	// check if we need to download the SVG from the media server
-	_, err := os.Stat(sm.getFullPath(mediaPath))
+	_, err := os.Stat(sm.getFullPath(customizedMediaPath))
 	if errors.Is(err, fs.ErrNotExist) {
 		svg, err := getCachedMediaResource(mediaPath)
 		if err != nil {
-			return err
+			return "", err
+		}
+
+		svg = customiseSnakeSVG(svg, c)
+
+		err = sm.writeFile(customizedMediaPath, []byte(svg))
+		if err != nil {
+			return "", err
+		}
+	}
+
+	// return the new media path which includes the colour directory
+	return customizedMediaPath, nil
+}
+
+// customiseSnakeSVG sets the fill colour for the outer SVG tag
+func customiseSnakeSVG(svg string, c color.Color) string {
```

Comment:
Note: I was worried that this might be slow so I also tried a really simple string manipulation alternative:
```go
rootSVGIdx := strings.Index(svg, "<svg")
	if rootSVGIdx == -1 {
		// no SVG tag, bail out
		return svg
	}

	// buf := bytes.NewBuffer(make([]byte, len(svg)+len(` fill="#000000"`)))
	var buf bytes.Buffer

	// write out the original SVG up until this point
	buf.WriteString(svg[:rootSVGIdx+4])
	svg = svg[rootSVGIdx+4:]

	// Find the end of the tag, so we can put the "fill" attribute there
	// This is in case there is already a fill attribute. The last one will win.
	for i, r := range svg {
		// closing part of tag found
		if r == '>' {
			buf.WriteString(fmt.Sprintf(` fill="%s"`, colorToHex6(c)))

			// and write the remaining part of the SVG
			buf.WriteString(svg[i:])
			break
		}

		// not at the closing tag yet
		buf.WriteRune(r)
	}

	return buf.String()
```

But when I ran some simple benchmarks I noticed that it didn't actually perform better. So we should probably stick with this XML parsing approach.